pool.go

```diff
@@ -314,6 +331,15 @@ func NewPool(network, addr string, size int, opts ...PoolOpt) (*Pool, error) {
 	if p.opts.overflowSize > 0 && p.opts.overflowDrainInterval > 0 {
 		p.atIntervalDo(p.opts.overflowDrainInterval, p.doOverflowDrain)
 	}
+
+	// wait all connections to establish - as tests show, you can't work with pool
+	// until p.initDone is closed!
```

Comment:
This is wrong. The Pool can be used before all connections are created. The benchmark just waits for the pool to be filled, to avoid measuring the overhead of creating connections.

Waiting for the pool to be full before returning here will also needlessly slow down many uses of Pools.

Also this is leaking the connections and goroutines.

Reply:
Thanks! Actually I've experienced some problems with tests when I tried to remove the wait on initDone.

I'll re-check it soon and come back with details.

Reply:
For clarity, _some_ tests do rely on the Pool to be full before they run, hence the need for initDone in the first place. But in general we don't want the Pool to be waiting for all connections to be created before returning.